tests/Doctrine/Tests/DBAL/Platforms/MySQL57PlatformTest.php

```diff
@@ -82,4 +82,22 @@ protected function getGeneratesAlterTableRenameIndexUsedByForeignKeySQL()
             'ALTER TABLE mytable RENAME INDEX idx_foo TO idx_foo_renamed',
         );
     }
+
+    /**
+     * @group DBAL-2576
+     */
+    public function testNotDuplicateDropForeignKeySql()
+    {
+        $diff = $this->createNotDuplicateDropForeignKeySql();
+
+        // Run through alter table method
+        $sql = $this->_platform->getAlterTableSQL($diff);
+
+        // Assert that there are no duplicates in the results
+        $this->assertSame([
+            'ALTER TABLE documents DROP FOREIGN KEY documents_ibfk_1',
```

Comment:
Given that the set of expected statements is different depending on the platform, I believe this feature should be tested in a functional way (statements are executed on a real DB) rather on the SQL level. This way, every DB platform will _prove_ the correctness of the generated statements.